### ISSUE TITLE
[1.10.2] Computers shutting down when chunks unload

### ISSUE THREAD

**Issue by mzx987** [MightyPirates/OpenComputers#2393]
First of all, I understand that this issue has been presented in both #2051 and #2169, but it seems to be happening only on the server-side.

In singleplayer, hooking a computer up with any form of power, leaving it, and coming back to it will still stay on. However, on our server, leaving and re-entering the chunk causes the computer's screen (if present) to freeze on what it was last showing, and shuts down the computer. I've even made sure that both the computer and the power source are within the same chunk.

I've tried using an Analyzer to see if any errors pop up, but everything is working fine. ~~The server also doesn't appear to be show anything out of the ordinary when this event occurs.~~ Scratch that. It's very rare, but sometimes a message pops up:
```
11:49:57 PM CONSOLE: thread/WARN] [OpenComputers]: Could not persist computer @ (-111.5, 57.5, 539.5).
11:49:57 PM CONSOLE: to literally persist userdata
```

The server's OS is Ubuntu Server 16.04.2 LTS.
Forge version 12.18.3.2281.
It is currently running the latest 1.10.2 build of OpenComputers (1.6.2.97).

If need be, I can present a list of mods that both the clients and the server are currently using, but I am not sure if that would help, since it works fine on singleplayer worlds.

**Comment by magik6k** [comment]
What components are connected to the computer? (check by running `components` command)

**Comment by mzx987** [comment]
Here is the list of components:
http://i.imgur.com/zGbN02N.png

Edit: Not sure why this would affect it. Regardless of component, they shut down anyway.

**Comment by TheYaMeZ** [comment]
I'm getting a similar issue I think.

As soon as the chunk unloads and I login/arrive at the chunk the computer is not on.
Checking with the analyzer does not give me any information and the server console does not have any clues either.

If it helps, my mod list is quite small too.
OpenComputers, ImmersiveEngineering, Pams Harvestcraft.
Also, i'm running the latest (as of time of writing) version of OC for 1.10.2

Please let me know if there is any other information that would be helpful

**Comment by mzx987** [comment]
@TheYaMeZ Is it happening to you on a server? If so, what OS is the server running on?

**Comment by SaphireLattice** [comment]
This seems to be quite a long existing issue. There should be few more
issues down there. I am not sure there's one universal solution though.

**Comment by TheYaMeZ** [comment]
@mzx987 I am indeed running on a server. I'm renting a server from "Minecraft Server Australia" who use multicraft. Other than that i'm not sure what it's running on but my guess would be Unix. Is there a console command that would give me more information?

**Comment by mzx987** [comment]
@TheYaMeZ Usually the server logs tell you what OS it's currently running on. Look at the "fml-server-latest.log" file and it should show in the first few lines.

Speaking of which, if it helps, my server logs tell me that it is running on Linux:amd64:4.4.0-78-generic with OpenJDK.

Edit: Although I checked this less than a week ago, I just want to confirm that it also detects and loads the native library that OpenComputers provides.

**Comment by mzx987** [comment]
Not sure if you are all gonna believe this: I recently installed SpongeForge today, and for some reason, persistence works just fine when SpongeForge and OpenComputers are together.

**Comment by TheYaMeZ** [comment]
>Usually the server logs tell you what OS it's currently running on. Look at the "fml-server-latest.log" file and it should show in the first few lines.

My logs indicate: "Linux:amd64:3.13.0-40-generic"

**Comment by mzx987** [comment]
As far as I can tell, after installing SpongeForge, nothing much else happens. But for some reason, the persistence occasionally works. I feel it's just a coincidence though.

**Comment by mzx987** [comment]
I really don't mean to sound impatient, but is there a fix for something like this? I really need persistent computers to work on this Linux server (the only workaround would be to use Chunk Loaders, but I don't want those spotted everywhere).

**Comment by DTV96Calibre** [comment]
Any news on this? We're using chunk loading and the computers still shut down when I leave the area. I'm willing to spend time trying to fix it if anyone knows anything about what could be causing it.

**Comment by skyem123** [comment]
I'm not sure how helpful it'd be, but maybe join the IRC channel and see if anyone can help you there?

@DTV96Calibre, it's probably also a good idea if you try to get any errors reported from the logs, to see if it's the same exact issue or something else, and Java version, OS version, OC version, etc etc helps, because while others have reported it, it might or might not be the same issue. Actually, the more people who report this with details, the eaiser it is to work out some common thing if it's all caused by the same thing.

Sorry I'm not much help here, the Lua code is kinda like black magic, only a few people here understand it... 😛 (It's C code and the mod is wrtiten in Scala, Minecraft is written in Java, and the stuff that runs on OC is Lua code...)

**Comment by DTV96Calibre** [comment]
I'm still waiting on getting the logs (a friend is hosting the server) but I have found a temporary solution that fixed my problem. I removed the internet card from my computer and it stopped mysteriously shutting down. Putting the card back in caused the issue to happen again, although I was unable to reproduce the bug on a fresh server I generated for testing it.

We're running FTB Beyond 1.9.0 which uses Minecraft 1.10.2 and OpenComputers 1.6.2.7. The server should be running on some recent version of Windows, but I can't yet say what version of Java is being used.

I'm just guessing but because I can't reproduce the issue on a server running on my personal machine (as opposed to the remote machine I'm connecting to), it might be a java version issue, though I'll be sure when I get a copy of the server to test on my personal machine.

**Comment by mzx987** [comment]
@DTV96Calibre, Odd. If I can recall correctly, an Internet Card _never_ did that to me. No matter what the configuration was, it would have issues staying persistent.

I would provide more information, but my server's been down for a while, so I can't really say anything else about this. However, if more information about this needs to be provided, I am more than willing to start it back up, update everything, and run it again.

**Comment by skyem123** [comment]
okay... I'll admit that it's really odd. I wonder if another world / MC server on the same physical server would have the same problem, or if it's the world being broken somehow...

**Comment by DTV96Calibre** [comment]
What I said about removing the internet card fixing the shutdown issue no longer seems to be true. The server was restarted and now there is no temporary solution for me.

**Comment by DTV96Calibre** [comment]
I think FTB Utilities is causing the issue. I tried using a chunk loading upgrade on a drone to keep my computer loaded and there was no crashing issue. I also tried running a computer outside of chunks claimed with FTB Utilities's Team GUI and again no crashing issue.

Edit: While this is true on the remote server I'm connecting to, I still can't reproduce the bug on a server running locally on my machine. I'll update this when I get a copy of the remote server and test it locally

**Comment by mzx987** [comment]
Well, my server is back up and has been running for a while. The issue still remains, from what I can see.

**Comment by payonel** [comment]
@mzx987 are you still running on 1.10?
are you still seeing this issue?
Can you please try http://ci.cil.li/job/OpenComputers-dev-MC1.10/lastSuccessfulBuild/artifact/build/libs/OpenComputers-MC1.10.2-1.6.2.122-dev.jar

**Comment by mzx987** [comment]
Yes, I am still running 1.10. I went ahead and installed the latest dev build.

I set up a Creative Computer in the middle of nowhere, and it appears to stay persistent. I'll do some more testing later on tomorrow to be sure.

**EDIT:** As a side-note, I had dev build 114 installed on the server before I updated.

**Comment by mzx987** [comment]
After leaving a computer to run on the server overnight (and several scheduled server resets later), it appears that the computer has remained persistent.

Not quite sure what has changed (may have been updates to Linux, or updates to OpenComputers), but I think it has been resolved (will attempt more testing later on). May need some others to test this issue out, though.

**Comment by payonel** [comment]
@mzx987 I'm closing this as fixed. We have specifically made improvements to both robots and computer loading issues (a crucial robot fix was the most recent).

If you find these issues again, please open a new ticket. Thank you
